### PR TITLE
fix an encoding issue with calling adb shell getprop

### LINKED ISSUE #5738: Crash when creating a new application
## Steps to Reproduce

On a Fedora 24 machine, while a Fire table is connected:
$ flutter create boyxing

## Flutter Doctor

It crashes with a similar error.

## Logs and Crash Reports

Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter create boyxing

## exception

FormatException: Bad UTF-8 encoding 0x8a

```
dart:io                                                           Process.runSync
package:flutter_tools/src/android/android_device.dart 65          AndroidDevice._getProperty
package:flutter_tools/src/android/android_device.dart 91          AndroidDevice.platform
package:flutter_tools/src/device.dart 252                         Device.descriptions.<fn>
dart:_internal                                                    ListIterable.toList
package:flutter_tools/src/doctor.dart 311                         DeviceValidator.validate.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/device.dart 70                          DeviceManager.getAllConnectedDevices.<async>
===== asynchronous gap ===========================
dart:async                                                        _Completer.completeError
package:flutter_tools/src/doctor.dart 315                         DeviceValidator.validate.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/device.dart 70                          DeviceManager.getAllConnectedDevices.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/doctor.dart                             DeviceValidator.validate
package:flutter_tools/src/doctor.dart 75                          Doctor.summaryText.summaryText_async_body>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/doctor.dart 265                         AtomValidator.validate.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/doctor.dart                             Doctor.summaryText
package:flutter_tools/src/doctor.dart 66                          Doctor.summary.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/doctor.dart                             Doctor.summary
package:flutter_tools/src/commands/create.dart 120                CreateCommand.runInProject.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/dart/pub.dart 60                        pubGet.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 82                    runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/commands/create.dart                    CreateCommand.runInProject
package:flutter_tools/src/runner/flutter_command.dart 192         FlutterCommand._run.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/cache.dart 184                          Cache.updateAll.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/runner/flutter_command.dart             FlutterCommand._run
package:flutter_tools/src/runner/flutter_command.dart 116         FlutterCommand.run
package:args/command_runner.dart 177                              CommandRunner.runCommand.<fn>
dart:async                                                        Future.Future.sync
package:args/command_runner.dart 130                              CommandRunner.runCommand
package:flutter_tools/src/runner/flutter_command_runner.dart 177  FlutterCommandRunner.runCommand.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/cache.dart 71                           Cache.lock.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/runner/flutter_command_runner.dart      FlutterCommandRunner.runCommand
package:args/command_runner.dart 104                              CommandRunner.run.<fn>
dart:async                                                        Future.Future.sync
package:args/command_runner.dart 104                              CommandRunner.run
package:flutter_tools/src/runner/flutter_command_runner.dart 123  FlutterCommandRunner.run
package:flutter_tools/executable.dart 94                          main.<async>.<fn>.<async>
===== asynchronous gap ===========================
package:stack_trace                                               Chain.capture
package:flutter_tools/executable.dart 88                          main.<async>
```

## flutter doctor

```
[✓] Flutter (on Linux, channel alpha)
    • Flutter at /home/alvaro/flutter-workspace/flutter
    • Framework revision a69c11f349 (2 days ago), 2016-09-02 17:01:11
    • Engine revision dcb026188a
    • Tools Dart version 1.19.0-dev.5.0

[✓] Android toolchain - develop for Android devices (Android SDK 24.0.1)
    • Android SDK at /home/alvaro/Android/Sdk
    • Platform android-24, build-tools 24.0.1
    • OpenJDK Runtime Environment (build 1.8.0_102-b14)

[✓] Atom - a lightweight development environment for Flutter
    • flutter plugin version 0.2.4
    • dartlang plugin version 0.6.35

[✓] Connected devices
    • KFASWI • 00DF06065183073J • android-arm
```

### PR DESCRIPTION
**Pull request by devoncarew**
- fix an encoding issue with calling adb shell getprop (don't assume UTF8)
- be defensive about exceptions from Process.runSync
- fix https://github.com/flutter/flutter/issues/5738

@eseidel @danrubel

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Can we test this?

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -62,13 +62,24 @@ class AndroidDevice extends Device {
...
+          propCommand.first,
+          propCommand.sublist(1),
+          stdoutEncoding: ASCII
+        );
+        if (result.exitCode == 0) {
+          RegExp propertyExp = new RegExp(r'\[(.*?)\]: \[(.*?)\]');
+          for (Match match in propertyExp.allMatches(result.stdout))
+            _properties[match.group(1)] = match.group(2);
+        } else {
+          printError('Error retrieving device properties for $name.');
```
Comment (alvaroc1) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Maybe this error message should include info from `result.stderr`?

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 2]
```diff
@@ -62,13 +62,22 @@ class AndroidDevice extends Device {
...
+        // `adb shell getprop` result as UTF8.
+        ProcessResult result = Process.runSync(
+          propCommand.first,
+          propCommand.sublist(1),
+          stdoutEncoding: ASCII
+        );
+        if (result.exitCode == 0) {
+          _properties = parseAdbDeviceProperties(result.stdout);
+        } else {
+          printError('Error retrieving device properties for $name.');
```
Comment (danrubel) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Is there any additional error information we should provide here? exitCode?

**Comment by danrubel** [comment]
LGTM with comment to address as you see fit.

**Comment by devoncarew** [comment]
Sounds like from the bug report that we'd need to update this to LATIN1 (even the ASCII encoding is getting interpreted and is failing).

**Comment by Hixie** [comment]
@devoncarew what's your plan with this PR?

**Comment by devoncarew** [comment]
Urk, this fell off my radar. I'll revisit, and fix it or close.

**Review by danrubel — APPROVED** [review]
Still LGTM

**Pull request merged**